tests/forte/data/converter/feature_test.py

```diff
@@ -0,0 +1,193 @@
+                                 [1, 1, 0, 0],
+                                 [1, 1, 1, 1],
+                                 [0, 0, 0, 0]]])
+
+        self.feature3.pad(4)
+        for sub_feature in self.feature3.sub_features:
+            sub_feature.pad(3)
+        feature, mask = self.feature3.data
+        self.assertEqual(feature, [[[0, 1, 0], [1, 0, 0], [1, 0, 0]],
+                                   [[1, 0, 0], [0, 1, 0], [0, 0, 1]],
+                                   [[0, 0, 1], [0, 0, 1], [0, 0, 1]],
+                                   [[0, 0, 1], [0, 0, 1], [0, 0, 1]]])
+        self.assertEqual(mask, [[1, 1, 0, 0],
+                                [[1, 1, 1],
+                                 [1, 1, 0],
+                                 [0, 0, 0],
+                                 [0, 0, 0]]])
+
+    def test_unroll_nopad(self):
+        feature, mask = self.feature1.data
+        self.assertEqual(feature, [7, 8, 9])
+
+        feature, mask = self.feature2.data
+        self.assertEqual(feature, [[6, 11, 2],
+                                   [7, 8],
+                                   [6, 7, 5, 4]])
+
+        feature, mask = self.feature3.data
+        self.assertEqual(feature, [[[0, 1, 0], [1, 0, 0], [1, 0, 0]],
+                                   [[1, 0, 0], [0, 1, 0]]])
+
+    def test_unroll_dtype(self):
+        self.feature1 = self.create_feature1(dtype=torch.float)
+        self.assertEqual(self.feature1.dtype, torch.float)
+
+        self.feature2 = self.create_feature2(dtype=torch.float)
+        self.assertEqual(self.feature2.dtype, torch.float)
+
+        for sub_feature in self.feature2.sub_features:
+            self.assertEqual(sub_feature.dtype, torch.float)
+
+    def create_feature1(self,
+                        data=None,
+                        pad_id=None,
+                        dim=None,
+                        dtype=torch.long):
+        data: List = [7, 8, 9] if data is None else data
+        pad_id: int = 0 if pad_id is None else pad_id
+        dim: int = 1 if dim is None else dim
+        feature: Feature = Feature(data, {"pad_value": pad_id,
+                                          "dim": dim,
+                                          "dtype": dtype
+                                         })
+
+        return feature
+
+    def create_feature2(self,
+                        data=None,
+                        pad_id=None,
+                        dim=None,
+                        dtype=torch.long):
+        data: List = [[6, 11, 2], [7, 8], [6, 7, 5, 4]] \
+            if data is None else data
+        pad_id: int = 0 if pad_id is None else pad_id
+        dim: int = 2 if dim is None else dim
+        feature: Feature = Feature(data, {"pad_value": pad_id,
+                                          "dim": dim,
+                                          "dtype": dtype
+                                         })
+
+        return feature
+
+    def create_feature3(self,
```

Comment:
Still expecting some feature dtype such as `str` in test cases

Reply:
Done